### PR TITLE
codec(ticdc): decoder support only handle key columns

### LINKED ISSUE #9189: decoder support decode message only have handle key columns

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #9189 

#### Tests 

 - Unit test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/sink/codec/decoder.go [hunk 1]
```diff
@@ -31,7 +31,9 @@ type RowEventDecoder interface {
 	// NextResolvedEvent returns the next resolved event if exists
 	NextResolvedEvent() (uint64, error)
 	// NextRowChangedEvent returns the next row changed event if exists
-	NextRowChangedEvent() (*model.RowChangedEvent, error)
+	// if the event only has handle key sent, return true,
+	// consumer should query the row data by the handle key and commit-ts
```
Comment (CharlesCheung96) [inline pkg/sink/codec/decoder.go:35]: Please add some comments to explain the differences in behavior betwee protocols.

**Review by CharlesCheung96 — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**